workflow/controller/operator.go

```diff
@@ -2023,9 +2024,9 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 		// and we return instead.
 		retryStrategy := woc.retryStrategy(processedTmpl)
 		if retryStrategy == nil ||
-			(retryStrategy.RetryPolicy != wfv1.RetryPolicyAlways &&
-				retryStrategy.RetryPolicy != wfv1.RetryPolicyOnError &&
-				retryStrategy.RetryPolicy != wfv1.RetryPolicyOnTransientError) {
+			(retryStrategy.RetryPolicyActual() != wfv1.RetryPolicyAlways &&
+				retryStrategy.RetryPolicyActual() != wfv1.RetryPolicyOnError &&
+				retryStrategy.RetryPolicyActual() != wfv1.RetryPolicyOnTransientError) {
 			woc.controller.syncManager.Release(woc.wf, node.ID, processedTmpl.Synchronization)
```

Comment:
Good idea to add a method here. Nitpick: can we call RetryPolicyActual() once here so we're not repeating calling the same logic over and over? (slight performance benefit, especially if that method were to get more complicated sometime in the future)

Reply:
Done